### PR TITLE
test(web): cover decodeBlobText across all five decoder branches

### LINKED ISSUE #19239: test(web): add unit tests for decodeBlobText (no coverage after #19222)
## Summary

`web/src/utils/file-util.ts:decodeBlobText` was added in #19222 to fix non-UTF-8 CSV/TXT preview encoding. It is now called from two preview paths:

- `web/src/components/document-preview/csv-preview.tsx` (CSV preview)
- `web/src/components/document-preview/txt-preview.tsx` (TXT preview)

The helper has **no unit tests** — `web/src/utils/__tests__/` covers `delimiter-preview`, `source-locate`, `chat`, `dataset-util`, `pipeline-operator`, but not `file-util`.

## Why it matters

`decodeBlobText` has five distinct branches:

1. UTF-16LE BOM (`0xff 0xfe`)
2. UTF-16BE BOM (`0xfe 0xff`)
3. UTF-8 BOM (`0xef 0xbb 0xbf`)
4. Strict UTF-8 (no BOM, valid UTF-8)
5. GBK fallback (when strict UTF-8 throws)

A regression in any of them would silently corrupt Chinese / multi-byte previews. Pinning the behavior with tests makes the contract explicit and prevents the BOM/GBK fallthrough from drifting during future refactors of the preview paths.

## Proposed tests

- UTF-8 with BOM: round-trip `Hello, 世界`; assert the leading `\ufeff` is stripped.
- UTF-16LE with BOM: `0xff 0xfe 0x48 0x00 0x69 0x00` → `"Hi"`.
- UTF-16BE with BOM: `0xfe 0xff 0x00 0x48 0x00 0x69` → `"Hi"`.
- Plain UTF-8 without BOM: `Hello, 世界` via `TextEncoder` round-trip.
- GBK fallback: `0xd6 0xd0 0xce 0xc4` → `"中文"`.
- Empty buffer → `""`.
- 1-byte buffer `0x41` → `"A"`.
- 2-byte buffer `0x41 0x42` → `"AB"` (no false BOM match).

## Scope

- One new file: `web/src/utils/__tests__/file-util.test.ts`.
- No production-code changes.
- Uses `jest` with `jsdom` + `node:util` `TextEncoder`/`TextDecoder` (already wired in `web/jest-setup.ts`).

## Acceptance criteria

- New tests pass: `cd web && pnpm jest src/utils/__tests__/file-util.test.ts --no-coverage`.
- Lint / format / type-check stay clean.
- Coverage for `file-util.ts` is no longer zero.

### PR DESCRIPTION
**Pull request by Harsh23Kashyap**
## Summary

`web/src/utils/file-util.ts:decodeBlobText` was added in #19222 to fix non-UTF-8 CSV/TXT preview encoding. After the fix, it is now called from two preview paths:

- `web/src/components/document-preview/csv-preview.tsx` (CSV preview)
- `web/src/components/document-preview/txt-preview.tsx` (TXT preview)

The helper has **no unit tests** — `web/src/utils/__tests__/` covers `delimiter-preview`, `source-locate`, `chat`, `dataset-util`, `pipeline-operator`, but not `file-util`.

This PR pins the contract: 8 tests across all five decoder branches the function takes (UTF-16LE BOM, UTF-16BE BOM, UTF-8 BOM, strict UTF-8, GBK fallback) plus three small-buffer edge cases (empty, 1-byte, 2-byte buffers) that exercise the `Math.min(3, buffer.byteLength)` slice and prevent a false BOM match on partial prefixes.

## Why it matters

`decodeBlobText` is a tiny helper, but it has five distinct branches, and a regression in any of them would silently corrupt Chinese / multi-byte previews. Pinned behavior prevents the BOM/GBK fallthrough from drifting during future refactors of the preview paths and gives the next maintainer a clear, executable specification.

## What changed

- New file: `web/src/utils/__tests__/file-util.test.ts` (92 lines, 8 tests, 1 commit).
- No production code change.

## Test design

The test passes a minimal Blob-substitute that exposes only `arrayBuffer()`, the single method `decodeBlobText` actually calls. This keeps the test independent of the jsdom Blob polyfill in this project's test environment (which does not implement `arrayBuffer()`); the production paths in `csv-preview.tsx` and `txt-preview.tsx` pass the real browser Blob returned by axios, so coverage of the real Blob is left to the E2E suite.

| Branch | Bytes | Expected |
|---|---|---|
| UTF-8 + BOM | `0xef 0xbb 0xbf` + UTF-8 `Hello, 世界` | `Hello, 世界` (BOM stripped) |
| UTF-16LE + BOM | `0xff 0xfe 0x48 0x00 0x69 0x00` | `Hi` |
| UTF-16BE + BOM | `0xfe 0xff 0x00 0x48 0x00 0x69` | `Hi` |
| Plain UTF-8 (no BOM) | `TextEncoder.encode('Hello, 世界')` | `Hello, 世界` |
| GBK fallback | `0xd6 0xd0 0xce 0xc4` (中文 in GBK) | `中文` |
| Empty buffer | `new Uint8Array(0)` | `""` |
| 1-byte | `0x41` | `A` |
| 2-byte non-BOM | `0x41 0x42` | `AB` |

## Verification

- `pnpm jest src/utils/__tests__/file-util.test.ts --no-coverage` → 8/8 pass.
- `pnpm oxfmt --check src/utils/__tests__/file-util.test.ts` (via npx) → clean.
- `pnpm oxlint src/utils/__tests__/file-util.test.ts src/utils/file-util.ts` (via npx) → no new warnings.
- `pnpm tsc --noEmit` → no new errors in this file (other unrelated pre-existing TS errors in the tree from missing `pnpm-lock.yaml` are not introduced by this change).

## Note on the review pipeline

`/consult-kiro` (the opencode-panel Tier A review) timed out on this run (the 13th consecutive timeout across cycles 49-62). I shipped on in-session review per the established workflow. The test file is short, declarative, and reviewed line-by-line before push; the contract is pinned, the production code is unchanged, and the diff is reviewable in under 30 seconds.

Fixes #19239

### CODE REVIEW

File: web/src/utils/__tests__/file-util.test.ts [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    // misclassifying it. Strict UTF-8 succeeds for plain ASCII.
+    const got = await decodeBlobText(blobOf(new Uint8Array([0x41])));
+    expect(got).toBe('A');
+  });
+
+  it('decodes a 2-byte buffer that is not a BOM prefix', async () => {
+    // 0x41 0x42 is "AB" in plain ASCII; bytes[0]=0x41, bytes[1]=0x42 does
+    // not match any of the three BOM checks (0xff 0xfe, 0xfe 0xff,
+    // 0xef 0xbb 0xbf), so the strict UTF-8 path is used.
+    const got = await decodeBlobText(blobOf(new Uint8Array([0x41, 0x42])));
```
[the remark this answers is not in the document]
Comment (Harsh23Kashyap) [reply web/src/utils/__tests__/file-util.test.ts]: Fixed in commit 247c8dd99 ("test(web): cover truncated UTF-8 BOM prefix in decodeBlobText"). Replaced the 0x41 0x42 fixture with 0xef 0xbb, which exercises the partial-BOM path: `Math.min(3, 2)` yields a 2-byte slice, `bytes[2]` is undefined and the BOM check fails (`undefined !== 0xbf`), strict UTF-8 throws on 0xbb as a stray continuation byte, and the GBK fallback decodes the 2 bytes as a single GBK double-byte character. The test pins structural properties only (non-empty, single character, no leading U+FEFF); the exact GBK-decoded code point is left implementation-defined. 8/8 pass on the full file.

**Review by cike8899 — APPROVED** [review]
